sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -447,6 +448,23 @@ pi_result getInfo<const char *>(size_t param_value_size, void *param_value,
                       param_value_size_ret, value);
 }
 
+std::string getKernelNames(pi_program program) {
+  std::string source(program->source_,
+                     program->source_ + program->sourceLength_);
+  std::regex entries_pattern(".entry\\s+([^\\([:s:]]*)");
+  std::string names("");
+  std::smatch match;
```

Comment:
Please, add tests.

Reply:
I have added a case for `has_kernel` in the `basic_test/kernel_info.cpp`. Currently that is the only place where this query is being used in SYCL and since the unit tests currently have no common mechanism for writing simple kernels for PI, this is probably the best we can do right now.

Reply:
Isn't [Unit tests](https://github.com/intel/llvm/tree/sycl/sycl/unittests/pi) good enough for that?

Reply:
The problem is adding unit tests with kernels, as would be needed for any sensible testing in here.

Reply:
Can we use clang build from sources to solve that "problem"? The same way LIT tests solve it.

Reply:
Good question. I'm not too familiar with it, but I think it would be beneficial to figure something out in general as the inability to use kernels in unit tests will soon be inhibiting the growth of the unit test set.